src/server/scene/basic_idle_hub.cpp

```diff
@@ -261,3 +264,16 @@ void ms::BasicIdleHub::schedule_alarm(ProofOfMutexLock const&, time::Timestamp c
         alarm_timeout = std::nullopt;
     }
 }
+
+void ms::BasicIdleHub::inhibit_idle()
+{
+    // Only the window in focus should be able to call inhibit_idle()
+    mir::log_info("Calling BasicIdleHub::inhibit_idle()");
+    alarm->cancel();
```

Comment:
First of all, this isn't calling `poke()` nor does it reimplement the part of poke that loops through and activates currently idle observers. This means if Mir has already hit any idle thresholds, those observer aren't woken up by the idle inhibit (which I think we want them to be).

Secondly. the alarm gets rescheduled every time the idle hub is poked (every input event). Therefore this should only function to not start the idle timeout until there is another input event. I think what's happening is the 0-timeout idle observer is preventing `BasicIdleHub` from working as intended (in a very roundabout way). 

My suggestion on how to fix this:
- add some way of checking if we are inhibited or not (such as a boolean, or locking a `weak_ptr` if you take my API suggestion)
- change `poke()` to only schedule the alarm if we are not inhibited
- call `poke()` after cancelling the alarm and marking the hub inhibited

That way any idle observers will be waken, future pokes wont schedule the alarm and you don't need the weird 0-duration timeout observer.